webknossos/webknossos/client/api_client/datastore_api_client.py

```diff
@@ -47,7 +50,7 @@ def dataset_reserve_upload(
         reserve_upload_information: ApiReserveDatasetUploadInformation,
         token: Optional[str],
         retry_count: int,
-    ) -> None:
+    ) -> httpx.Response:
```

Comment:
I think I’d prefer not to return the response here. The callers aren’t supposed to do anything with it. In case of an error response, an exception is already thrown. So unless you have a special reason, please change back the return type and instead remove the `return` statement below. This should also make the httpx import unnecessary

Reply:
Alright, I removed the `return` statements. Previously, some methods like `dataset_finish_upload` returned the result of `_post_json` which always was `None`. I thought it would make sense to either make `_post_json` return something or remove the return statement from `dataset_finish_upload` and `dataset_reserve_upload`.